packages/frontend/src/api/rpc.ts

```diff
@@ -41,16 +42,51 @@ export function createRpcClient(serverUrl: string, firebaseApp?: FirebaseApp) {
     return build_client<QubitServer>(transport);
 }
 
-/** Gets the content an RPC result, if it is `Ok`. */
-export function resultOk<T>(result?: RpcResult<T>): T | undefined {
-    return result?.tag === "Ok" ? result.content : undefined;
+// Type guard for type narrowing inside SolidJS JSX blocks controlled by a `when` attribute.
+//
+// example usage:
+// type Result = { tag: "Ok", content: string } | { tag: "Err", code: number };
+// const [result, setResult] = createSignal<Result>({tag: "Err", code: 0 });
+// ...
+// <Match when={narrow(result, (res) => res.tag === "Ok")}>
+//     {(result) => {
+//         const a = result(); // the type of a is narrowed to { tag: "Ok", content: string }
+// }
+export function narrow<A, B extends A>(accessor: Accessor<A>, guard: (v: A) => v is B): B | null {
```

Comment:
Am I correct that this function is included for pedagogy and not intended to be used directly? If so, while I appreciate the explanation, I don't think we should leave it in the codebase on the general principle that we shouldn't have "dead code."

The docstring is useful though: I'd suggest adding an example usage in this style to `rpcResourceOk` and/or `rpcResourceErr`.

Reply:
> Am I correct that this function is included for pedagogy and not intended to be used directly?

Not entirely, it can be used directly and I was doing so initially. I created the specialised `rpcResourceOk` and `rpcResourceErr` functions for readability and possible future reuse.

It's not immediately obvious how to write the more general `narrow` function if given only the specialised cases, and it took a non-trivial amount of frustration to figure out what was wrong and to find a solution. 

I propose that the reduction in possible future frustration is worth a couple of dead lines of code. However it's not obvious to me if the same trade off is worth it if it means a breaking a previously well enforced rule.

Reply:
Ok, since `narrow` is itself usable/has been used before, I'm happy to leave it in.